lib/leanplum_api/api.rb

```diff
@@ -20,6 +20,38 @@ def set_device_attributes(device_attributes, options = {})
       track_multi(device_attributes: device_attributes, options: options)
     end
 
+    # POSTs to Leanplum's sendMessage API endpoint
+    #
+    # @param message_id [String] the Leanplum message ID
+    # @param user_id [String] the Leanplum user ID
+    # @param device_id [String] the Leanplum device ID
+    # @param create_disposition [String] the policy that determines whether users are created by the API.
+    # @param force [Boolean] whether to send the message regardless of whether the user meets the targeting criteria.
+    # @param values [Hash{Symbol => String, Numeric}] values used to set variables used in the message.
+    # @return [Array<Hash>] the Response(s) from the API
+    def send_message(message_id:, user_id:, device_id: nil, create_disposition: "CreateNever", force: false, values: {}, dev_mode: false)
+      message = build_send_message(message_id: message_id, user_id: user_id, device_id: device_id, create_disposition: create_disposition, force: force, values: values, dev_mode: dev_mode)
+      production_connection.multi([message]).body['response']
+    end
+
+    # POSTs multiple messages to Leanplum's sendMessage API endpoint
+    #
+    # @param message_id [String] the Leanplum message ID
+    # @param users [Array<Hash{:id => String (required), :device_id => String (optional), :values => Hash (optional)}>] Leanplum users including an optional device_id and vaules hash used to set the variables for the message template
+    # @param create_disposition [String] the policy that determines whether users are created by the API.
+    # @param force [Boolean] whether to send the message regardless of whether the user meets the targeting criteria.
+    # @param values [Hash{Symbol => String, Numeric}] values used to set variables used in the message.
+    # @return [Array<Hash>] the Response(s) from the API
+    def send_messages(message_id:, users:, create_disposition: "CreateNever", force: false, values: {}, dev_mode: false)
+      fail ArgumentError, "Cannot accept more than 50 users due to Leanplum API restrictions" if users.size > 50
```

Comment:
move this inside the validate_users
Also need a present? check before you call a size check.